packages/react-server/core/context/Navigator.js

```diff
@@ -161,7 +161,10 @@ class Navigator extends EventEmitter {
 
 		// call page.handleRoute(), and use the resulting code to decide how to
 		// respond.
-		page.handleRoute().then(handleRouteResult => {
+		// We call it in a promise handler so any exception that
+		// arises will get converted to a rejection that we can handle
+		// below.
+		Q().then(page.handleRoute).then(handleRouteResult => {
```

Comment:
Ah, um, hmm, weird. I _swear_ that I tried this exact thing before I wrote that message earlier, and it didn't resolve the promise.
```
> var foo = () => { return {foo: "bar"} };
undefined
> Q().then(foo).then(x => console.log(x))
```
But I just did it again, and it resolved so... :shrug:

LGTM :+1: